docs/manual/java/guide/services/AdditionalRouters.md

```diff
@@ -0,0 +1,132 @@
+# Additional Routers
+
+Since Lagom 1.5.0, is it possible to extend a Lagom Service with additional Play Routers. 
+
+This is particular useful when itegrating Lagom with existing Play Routers, for instance a [Play gRPC Router](https://developer.lightbend.com/docs/play-grpc/0.5.0/lagom/serving-grpc.html) , or any other Play router that you have at your disposal.
+
+In Java, you add an additional router when binding your Lagom Service in your Guice module. You should pass the additonal routers to the `bindService` method together with the Service interface and its implementation. You do this by means of a helper method called `additonalRouters`. 
+
+There are two variants for `additionalRouters`, one that receives a `Class<play.api.routing.Router>` and one that receives a `play.api.routing.Router`.
+
+The first one should be used when your Router has some dependencies and needs to get them injected  by Lagom's runtime DI infrastructure (Guice). In this case, you pass the class and Guice will initialize it with the right dependencies.
+
+
+```java
+public class HelloWorldModule extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindService(
+                HelloService.class, HelloServiceImpl.class,
+                additionalRouter(SomeOtherPlayRouter.class)
+        );
+    }
+}
+```
+
+The second variant should be used be used when the Router does not have any other dependencies and therefore can be immediately passed as an instance.
+
+```java
+public class HelloWorldModule extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindService(
+                HelloService.class, HelloServiceImpl.class,
+                additionalRouter(new SomeOtherPlayRouter())
+        );
+    }
+}
+```
+
+
+
+## File Upload Example
+
+The following example shows how you can add a file upload endpoint to an existing Lagom Service. 
+
+The example is based on [JavaRoutingDsl](https://www.playframework.com/documentation/2.7.x/JavaRoutingDsl) that allows you to build a Play Router programmatically. It adds an extra path (`/api/files`) that receives POST calls for multipart-form data. 
+
+```java
+class FileUploadRouter implements SimpleRouter {
+
+    private final Router delegate;
+
+    @Inject
+    public FileUploadRouter(RoutingDsl routingDsl) {
+        this.delegate = routingDsl
+                .POST("/api/files")
+                .routingTo( __ -> {
```

Comment:
TIL `__`

Reply:
I learn it a few days back on a akka PR. Single underscore is illegal, but double works! ;-)